fastfood/food.py

```diff
@@ -42,40 +41,35 @@ def build_cookbook(build_config, templatepack, cookbooks_home,
     with open(build_config) as cfg:
         cfg = json.load(cfg)
 
-    cookbook = None
-    if cookbook_path:
-        try:
-            cookbook = book.CookBook(cookbook_path)
-        # change this to a more specific error
-        except ValueError:
-            # create a new one please
-            pass
-
+    cookbook_name = cfg['name']
     written_files = []
-    if not cookbook:
-        cookbook_name = cfg['name']
-        files, cookbook = create_new_cookbook(
-            cookbook_name, templatepack, cookbooks_home, force=force)
-        written_files.extend(files)
+    cookbook = create_new_cookbook(cookbook_name, cookbooks_home)
 
     for stencil in cfg['stencils']:
         stencil_set = stencil.pop('stencil_set')
         # items left un-popped are **options
-        files, updated_cookbook = update_cookbook(
-            cookbook, templatepack, stencil_set, **stencil)
+        files, updated_cookbook = process_stencil(
+            cookbook,
+            cookbook_name,  # pass this in case metadata.rb isn't written yet
```

Comment:
Ohh. I just realized that `create_new_cookbook` does not do what I thought it did. I thought it would basically do the equivalent of `fastfood build` on the `base` stencil.